x/amm/keeper/update_pool_for_swap.go

```diff
@@ -157,6 +158,15 @@ func (k Keeper) UpdatePoolForSwap(
 
 	k.SetPool(ctx, pool)
 
+	// convert the fees into USD
+	swapFeeValueInUSD := k.ConvertCoinsToUsdcValue(ctx, swapFeeInCoins).String()
+	slippageAmountInUSD := k.CalculateUSDValue(ctx, tokenIn.Denom, sdkmath.Int(slippageAmount)).String()
+	weightRecoveryFeeAmountInUSD := k.CalculateUSDValue(ctx, tokenIn.Denom, weightRecoveryFeeAmount).String()
+	bonusTokenAmountInUSD := k.CalculateUSDValue(ctx, tokenOut.Denom, bonusTokenAmount).String()
+
+	// emit swap fees event
+	types.EmitSwapFeesCollectedEvent(ctx, swapFeeValueInUSD, slippageAmountInUSD, weightRecoveryFeeAmountInUSD, bonusTokenAmountInUSD)
```

Comment:
@amityadav0 shall we have this when single side exit in leverage lp?

Reply:
yes, and in join pool too @99adarsh